core/services/relay/evm/evm.go

```diff
@@ -218,21 +218,38 @@ func NewRelayer(ctx context.Context, lggr logger.Logger, chain legacyevm.Chain,
 	}
 
 	wCfg := chain.Config().EVM().Workflow()
-	// Initialize write target capability if configuration is defined
-	if wCfg.ForwarderAddress() != nil && wCfg.FromAddress() != nil {
-		if wCfg.GasLimitDefault() == nil {
-			return nil, fmt.Errorf("unable to instantiate write target as default gas limit is not set")
-		}
 
-		capability, err := NewWriteTarget(ctx, relayer, chain, *wCfg.GasLimitDefault(), lggr)
-		if err != nil {
-			return nil, fmt.Errorf("failed to initialize write target: %w", err)
+	// Do not register write target without required configuration
+	if wCfg.ForwarderAddress() == nil || wCfg.GasLimitDefault() == nil {
+		return relayer, nil
+	}
+
+	fromAddress := wCfg.FromAddress()
+
+	// If fromAddress is not specified, default to the eth key if there is only one
+	if fromAddress == nil {
+		ethKeys, err2 := opts.CSAETHKeystore.Eth().GetAll(ctx)
+		if err2 != nil {
+			return nil, fmt.Errorf("failed to get all eth keys: %w", err2)
 		}
-		if err := relayer.capabilitiesRegistry.Add(ctx, capability); err != nil {
-			return nil, err
+
+		// Do not register write target if there are multiple or no eth keys
+		if len(ethKeys) > 1 || len(ethKeys) == 0 {
+			return relayer, nil
```

Comment:
How about we panic here? This is a branch where forwarder is non-nil, fromAddress is nil and we have multiple keys - should be OK to break in the loudest possible way.